### PR TITLE
[INLONG-4301][Manager] Support oracle source

### LINKED ISSUE #4301: [Feature][Manager] Support oracle source
### Description

Support oracle source 

### Use case

_No response_

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by LvJiancheng**
### Title Name:[INLONG-4301][Manager]Support oracle source

where *XYZ* should be replaced by the actual issue number.

Fixes #4301 

### Motivation

Support oracle source

### Modifications

Add some utils and objects to the adaptative oracle data source in sort

- [X] This change added tests and can be verified as follows:

  *(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

  - Does this pull request introduces a new feature? (yes)
  - If yes, how is the feature documented? (JavaDocs)

### CODE REVIEW

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/sort/util/ExtractNodeUtils.java [hunk 1]
```diff
@@ -280,4 +284,44 @@ public static PostgresExtractNode createExtractNode(PostgresSourceResponse postg
...
+                properties,
+                primaryKey,
+                hostName,
+                userName,
+                password,
+                database,
+                schemaname,
+                tablename,
+                port,
+                null
```
Comment (yunqingmoswu) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/sort/util/ExtractNodeUtils.java:324]: Make sure it is null all right?

File: inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/enums/SourceType.java [hunk 2]
```diff
@@ -42,6 +43,7 @@ public enum SourceType {
     public static final String SOURCE_KAFKA = "KAFKA";
     public static final String SOURCE_PULSAR = "PULSAR";
     public static final String SOURCE_POSTGRES = "POSTGRES";
+    public static final String SOURCE_ORACLE = "oracle";
```
Comment (yunqingmoswu) [inline inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/enums/SourceType.java:46]: oracle->ORACLE?

File: inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/enums/SourceType.java [hunk 3]
```diff
@@ -42,6 +43,7 @@ public enum SourceType {
...
```
Comment (gong) [inline inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/enums/SourceType.java:47]: uppercase letter

**Comment by gong** [comment]
please some UT

File: inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/enums/SourceType.java (hunk 2 above)
Comment (LvJiancheng) [reply inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/enums/SourceType.java]: done

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/sort/util/ExtractNodeUtils.java (hunk 1 above)
Comment (LvJiancheng) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/sort/util/ExtractNodeUtils.java]: good idea

**Review by gong — APPROVED** [review]

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/sort/util/ExtractNodeUtils.java [hunk 4]
```diff
@@ -283,4 +288,56 @@ public static PostgresExtractNode createExtractNode(PostgresSource postgresSourc
...
+        final String name = oracleSource.getSourceName();
+        final String database = oracleSource.getDatabase();
+        final String schemaname = oracleSource.getSchemaName();
+        final String tablename = oracleSource.getTableName();
+        final String primaryKey = oracleSource.getPrimaryKey();
+        final String hostName = oracleSource.getHostname();
+        final String userName = oracleSource.getUsername();
+        final String password = oracleSource.getPassword();
+        final Integer port = oracleSource.getPort();
+        final String scanstartupMode = oracleSource.getScanStartupMode();
```
Comment (healchow) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/sort/util/ExtractNodeUtils.java:309]: Please use the lower camel format for variables.

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/sort/util/ExtractNodeUtils.java [hunk 5]
```diff
@@ -283,4 +288,56 @@ public static PostgresExtractNode createExtractNode(PostgresSource postgresSourc
...
+        OracleConstant.ScanStartUpMode scanStartupMode;
+        switch (scanstartupMode) {
+            case "initial":
```
Comment (healchow) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/sort/util/ExtractNodeUtils.java:312]: Please use some constant or enum to replace those magic strings.

**Review by healchow — CHANGES_REQUESTED** [review]

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/sort/util/ExtractNodeUtils.java (hunk 4 above)
Comment (LvJiancheng) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/sort/util/ExtractNodeUtils.java]: done

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/sort/util/ExtractNodeUtils.java (hunk 5 above)
Comment (LvJiancheng) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/sort/util/ExtractNodeUtils.java]: done

**Review by healchow — APPROVED** [review]

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/sort/util/ExtractNodeUtils.java [hunk 6]
```diff
@@ -283,4 +288,45 @@ public static PostgresExtractNode createExtractNode(PostgresSource postgresSourc
...
+                null : ScanStartUpMode.forName(oracleSource.getScanStartupMode());
+        List<StreamField> streamFieldInfos = oracleSource.getFieldList();
+        final List<FieldInfo> fieldInfos = streamFieldInfos.stream()
+                .map(streamFieldInfo -> FieldInfoUtils.parseStreamFieldInfo(streamFieldInfo, name))
+                .collect(Collectors.toList());
+        Map<String, String> properties = Maps.newHashMap();
+        return new OracleExtractNode(id,
+                name,
+                fieldInfos,
+                null,
```
Comment (yunqingmoswu) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/sort/util/ExtractNodeUtils.java:319]: Why it is null?

File: inlong-manager/manager-service/src/test/java/org/apache/inlong/manager/service/core/source/OracleSourceServiceTest.java [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+
+/**
+ * Oracle source service test
+ */
+public class OracleSourceServiceTest extends ServiceBaseTest {
+
+    private final String sourceName = "stream_source_service_test";
+    private static final String hostname = "127.0.0.1";
+    private static final Integer port = 1251;
```
[the remark this answers is not in the document]
Comment (LvJiancheng) [reply inlong-manager/manager-service/src/test/java/org/apache/inlong/manager/service/core/source/OracleSourceServiceTest.java:39]: done

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/sort/util/ExtractNodeUtils.java (hunk 6 above)
Comment (LvJiancheng) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/sort/util/ExtractNodeUtils.java:319]: now field watermarkField Unity remains empty in other data sources ，so，，

File: inlong-sort/sort-common/src/main/java/org/apache/inlong/sort/protocol/constant/OracleConstant.java [hunk 8]
```diff
@@ -111,6 +113,15 @@ public enum ScanStartUpMode {
         ScanStartUpMode(String value) {
             this.value = value;
         }
+
+        public static ScanStartUpMode forName(String name) {
+            for (ScanStartUpMode dataType : ScanStartUpMode.values()) {
+                if (dataType.getValue().equals(name.toLowerCase(Locale.ROOT))) {
```
[the remark this answers is not in the document]
Comment (LvJiancheng) [reply inlong-sort/sort-common/src/main/java/org/apache/inlong/sort/protocol/constant/OracleConstant.java]: en,it has changed case sensitivity

**Review by yunqingmoswu — APPROVED** [review]

**Pull request merged**